mosu-app/src/features/auth-signup/ui/StepNavigator.tsx

```diff
@@ -2,21 +2,21 @@ import { ChevronRight } from "lucide-react";
 
 import { cn } from "@/shared/lib/utils";
 
-import { useSignUpContext } from "../contexts/SignUpContext";
+const signupSteps = ["signup-step1", "signup-step2", "signup-step3"] as const;
 
-export const SignUpStepNavigator = () => {
-    const step = useSignUpContext().state.step;
+export const SignUpStepNavigator = ({ step }: { step: string }) => {
+    const stepIndex = signupSteps.indexOf(step as (typeof signupSteps)[number]) + 1;
 
     return (
         <nav>
             <ul className="flex h-[48px] w-full items-center justify-between text-sm font-semibold">
-                <SignInStepNavigatorItem index={1} label="약관 동의" active={step >= 1} />
+                <SignInStepNavigatorItem index={1} label="약관 동의" active={stepIndex >= 1} />
```

Comment:
stepIndex에 특정 의미를 부여해서 상수화 하는건 어떨까요?
```suggestion
                <SignInStepNavigatorItem index={1} label="약관 동의" active={stepIndex >= AGREEMENT_PHASE} />
```